targets/TARGET_STM/mbed_overrides.c

```diff
@@ -129,11 +129,14 @@ void mbed_sdk_init()
         LL_HSEM_1StepLock(HSEM, CFG_HW_STOP_MODE_SEMID);
         /*Release HSEM in order to notify the CPU2(CM4)*/
         LL_HSEM_ReleaseLock(HSEM, CFG_HW_STOP_MODE_SEMID, 0);
-    } else {
+    }
+#ifndef NO_FORCED_CM4_BOOT
```

Comment:
Do you have a specific Option Byte configuration ?

Reply:
Yes, our configuration is SYSCFG_UR1_BCM4 = 0 but we don't want to force the CM4 boot (because it could not contain any valid binary)

Reply:
If CM4 doesn't contains valid binary then switch `DUAL_CORE `should be disabled, and thus no need for `NO_FORCED_CM4_BOOT`

Reply:
Not so easy as DUAL_CORE is define at CMSIS level in stm32h747xx.h

Reply:
@jeromecoutant , is there no mbed switch to define dual core usage ?

Reply:
@facchinm, after discussion with @jeromecoutant:
As there is currently no real dual core management in mbed, we agree to add this new compialtion switch NO_FORCED_CM4_BOOT
Nevertheless, in order to have the same behavior (like your example where CM4 runs only when RPC lib requires it ), whatever OptionByte BOOT_CM4, we propose to move `#ifndef NO_FORCED_CM4_BOOT` few lines above, just before the`if` so that both `if `and `else `cases are covered

```
#ifndef NO_FORCED_CM4_BOOT
   if (READ_BIT(SYSCFG->UR1, SYSCFG_UR1_BCM4)) {
        LL_HSEM_1StepLock(HSEM, CFG_HW_STOP_MODE_SEMID);
        /*Release HSEM in order to notify the CPU2(CM4)*/
        LL_HSEM_ReleaseLock(HSEM, CFG_HW_STOP_MODE_SEMID, 0);
    }
```

Reply:
Macro name could be, then, something like `CM4_BOOT_BY_APPLICATION` ?

Reply:
@jeromecoutant @ABOSTM I agree on changing the define but I'm unsure about broadening the scope of the `ifndef`.
Let me expand:
if you leave BCM4 bit == 1 I presume you want to automatically boot the M4 too, so it makes sense to synchronize the processors (otherwise the M4 would boot with garbage in any case).
The other case is a bit more convoluted, so the user may or may not have code on the M4 and boot it via the application.

Reply:
I understand your point but relying on Option bytes to adapt software behavior suppose Hardware settings, Option Bytes, are in line with Software application development, and are set on purpose.
That is probably the case when you develops your application, but there may be users that are not aware of their Option byte settings, or just keep default , or chnage then for test, debug accidentally, ...
Mbed is a generic framework, we don't know which application will be developed, we cannot guarrante OpionBytes settings. So I prefer to avoid relying on hypothetical alignment between hardware and software. Instead I prefer to have the same behavior whatever Option Bytes setting. This is simpler for beginner user, and may prevent misalignment problem. 
In other words, Software configuration (new switch CM4_BOOT_BY_APPLICATION) take the precedence on hardware settings (Option Bytes).

From my point of view, STM32H747 in Mbed today doesn't tackle the case of no binary in CM4, and even less the case where M4 binary is loaded after boot by CM7.  And I am not sure we want to go this way.

Reply:
Agree! I reworked the commits to apply both the changes